tests/terraform/graph/checks/test_yaml_connected_nodes.py

```diff
@@ -48,17 +50,45 @@ def test_S3BucketLogging_connected_node(self):
         assert report.passed_checks[1].connected_node['resource'] == 'aws_s3_bucket_logging.example'
         assert report.passed_checks[1].connected_node['file_line_range'] == [14, 19]
 
-    def get_report(self, dir_name):
+    def test_correct_connected_node_per_check(self):
+        failed_check_id_with_connected_node = 'CKV_AWS_144'
+        pass_check_id_with_connected_node = 'CKV_AWS_145'
+        pass_check_id_without_connected_node = 'CKV_AWS_20'
+        check_ids = [failed_check_id_with_connected_node, pass_check_id_with_connected_node, pass_check_id_without_connected_node]
+
+        for graph_framework in ["NETWORKX", "RUSTWORKX"]:
+            report = self.get_report("connected_nodes", checks_ids=check_ids, graph_framework=graph_framework)
+            assert len(report.passed_checks) == 2
+            assert len(report.failed_checks) == 1
+
+            assert report.failed_checks[0].connected_node['resource'] == 'aws_s3_bucket_replication_configuration.replication'
+            assert report.failed_checks[0].connected_node['file_path'] == '/bla2.tf'
+            assert report.failed_checks[0].connected_node['file_line_range'] == [6, 18]
+
+            passed_with_connection = next((c for c in report.passed_checks if c.check_id == pass_check_id_with_connected_node), None)
+            passed_without_connection = next((c for c in report.passed_checks if c.check_id == pass_check_id_without_connected_node), None)
+
+            assert passed_with_connection.connected_node is not None
+            assert passed_without_connection.connected_node is None
+
+            assert passed_with_connection.connected_node['resource'] == 'aws_s3_bucket_server_side_encryption_configuration.bad_sse'
+            assert passed_with_connection.connected_node['file_path'] == '/bla2.tf'
+            assert passed_with_connection.connected_node['file_line_range'] == [20, 29]
+
+
+    def get_report(self, dir_name, graph_framework='NETWORKX', checks_ids=None):
         dir_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
                                 f"resources/{dir_name}")
         assert os.path.exists(dir_path)
         policy_dir_path = os.path.dirname(checks.__file__)
         assert os.path.exists(policy_dir_path)
-        for root, _, f_names in os.walk(policy_dir_path):
-            for f_name in f_names:
-                if f_name != f"{dir_name}.yaml":
-                    continue
-                policy = load_yaml_data(f_name, root)
-                assert policy is not None
-                with mock.patch.dict('os.environ', {'CHECKOV_GRAPH_FRAMEWORK': 'NETWORKX'}):
-                    return get_policy_results(dir_path, policy)
+        with mock.patch.dict('os.environ', {'CHECKOV_GRAPH_FRAMEWORK': graph_framework}):
```

Comment:
please refer to other places in the test which do that, we usually use a decorator on top of the function

Reply:
done, the issue was that the test was inside the unittest class which prevents the `pytest.mark.parametrize` behavior. ended up moving it to outside of the class